src/modes/direct_select.js

```diff
@@ -165,7 +165,11 @@ DirectSelect.toDisplayFeatures = function(state, geojson, push) {
 };
 
 DirectSelect.onTrash = function(state) {
-  state.selectedCoordPaths.sort().reverse().forEach(id => state.feature.removeCoordinate(id));
+  // Uses number-aware sorting to make sure '9' < '10'. Comparison is reversed because we want them
+  // in reverse order so that we can remove by index safely.
+  state.selectedCoordPaths
+    .sort((a, b) => b.localeCompare(a, 'en', { numeric: true }))
```

Comment:
```suggestion
    .sort((a, b) => +b - a)
```

Reply:
This doesn't work for sort keys that are deeply nested, e.g. `0.8.9` should be sorted before `0.8.10`

Reply:
I tested that this works with localeCompare's numeric sort:

```js
console.warn(['11', '10', '9', '33', '8', '30', '3'].sort((a, b) => b.localeCompare(a, 'en', { numeric: true })));
console.warn(['4.11', '4.10', '40.8', '4.9', '4.33',].sort((a, b) => b.localeCompare(a, 'en', { numeric: true })));
console.warn(['4.10.34', '4.10.3', '4.10.29'].sort((a, b) => b.localeCompare(a, 'en', { numeric: true })));
```

Reply:
Oh, I didn't realize those weren't just numbers, which wasn't clear from the comment. 👍